control-plane-operator/controllers/hostedcontrolplane/hostedcontrolplane_controller.go

```diff
@@ -905,6 +926,32 @@ func (r *HostedControlPlaneReconciler) update(ctx context.Context, hostedControl
 		errs = append(errs, err)
 	}
 
+	if cpoV2() {
+		// reconcile components only when kube apiserver is fully ready
+		// TODO(Mulham): remove when dependcnies managment abstraction is implemented.
+		kubeAPIServerDeployment := manifests.KASDeployment(hostedControlPlane.Namespace)
+		if err := r.Get(ctx, client.ObjectKeyFromObject(kubeAPIServerDeployment), kubeAPIServerDeployment); err == nil {
+			if util.IsDeploymentReady(ctx, kubeAPIServerDeployment) {
+				cpContext := component.ControlPlaneContext{
+					Context:                   ctx,
```

Comment:
that would be cool, but I think it makes it a bit difficult to figure out what fields in `ControlPlaneContext` are available to use. you would have to look for all `FromContext` functions to find if something you need is there, rather than just looking at the struct fields.